python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -26,15 +26,14 @@ class UPathIOManager(MemoizableIOManager):
 
     """
 
-    extension: str = ""  # override in child class
+    extension: Optional[str] = None  # override in child class
```

Comment:
Why change to `None` instead of `""`?

Reply:
Just reflecting general practice throughout the codebase-- when other strings are null, we use `None`, when list params are null, we use `None` instead of empty list etc. Using an `""` for a null value sticks out.

Reply:
Got it. It sounds like a good change, but I think it should be in a separate PR unless it's part of making the cloud IO managers use the UPathIOManager.

Reply:
Hmm-- in that case I'd reframe this PR then as a general refactor of `UPathIOManager` with the aim of making it usable as a base for the cloud IO managers. IMO that seems like an appropriate scope for a PR. `UPathIOManager` basically gets overhauled here.

Reply:
Got it - I think that for whatever reason this part of the change is making my spidey-sense tingle more than the other changes in this PR.    E.g. I don't think we have any formal guarantees here, but if someone extended UPathIOManager and was depending on the type of `extension` to not be `None`, this would break them.

Isolating it would have made it easier for me to approve this PR earlier because I wouldn't have needed to think through the implications.

That said, not a big deal.